From 7649d0c93dddfa6969e432c2510379aa5cf70420 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 4 Mar 2021 16:36:05 +0100
Subject: [PATCH] build: Lower mutter requirement

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 66c0fc073..5654a8838 100644
--- a/meson.build
+++ b/meson.build
@@ -25,7 +25,7 @@ gio_req = '>= 2.56.0'
 gi_req = '>= 1.49.1'
 gjs_req = '>= 1.65.1'
 gtk_req = '>= 3.15.0'
-mutter_req = '>= 40.rc'
+mutter_req = '>= 40.beta'
 polkit_req = '>= 0.100'
 schemas_req = '>= 3.33.1'
 startup_req = '>= 0.11'
-- 
2.30.1

