diff --git a/js/ui/overview.js b/js/ui/overview.js
index 78736e513eb69f3f3516b53bc3be68b9290fab3f..f7b07ebd3cde0b5764ee6d4b7eb60d60da83cb48 100644
--- a/js/ui/overview.js
+++ b/js/ui/overview.js
@@ -155,6 +155,10 @@ var Overview = class {
         return this._visibleTarget;
     }
 
+    get closing() {
+        return this._animationInProgress && !this._visibleTarget;
+    }
+
     _createOverview() {
         if (this._overview)
             return;
@@ -625,7 +629,6 @@ var Overview = class {
         this._visible = false;
         this._animationInProgress = false;
 
-        this.emit('hidden');
         // Handle any calls to show* while we were hiding
         if (this._shown)
             this._animateVisible();
@@ -635,6 +638,8 @@ var Overview = class {
         Main.panel.style = null;
 
         this._syncGrab();
+
+        this.emit('hidden');
     }
 
     toggle() {
diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index 34c54b61e11c3cc1677f59c12e0fe45c1d06e6df..3aa111017de2f7f31d66e1cf5ccc0afd5afa9f24 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -1122,7 +1122,8 @@ var WindowManager = class {
     }
 
     _shouldAnimate() {
-        return !(Main.overview.visible || this._workspaceAnimation.gestureActive);
+        const overviewOpen = Main.overview.visible && !Main.overview.closing;
+        return !(overviewOpen || this._workspaceAnimation.gestureActive);
     }
 
     _shouldAnimateActor(actor, types) {
@@ -1449,6 +1450,17 @@ var WindowManager = class {
         dimmer.setDimmed(false, this._shouldAnimate());
     }
 
+    _easeAfterOverview(actor, props) {
+        if (Main.overview.visible) {
+            let id = Main.overview.connect('hidden', () => {
+                actor.ease(props);
+                Main.overview.disconnect(id);
+            });
+        } else {
+            actor.ease(props);
+        }
+    }
+
     _mapWindow(shellwm, actor) {
         actor._windowType = actor.meta_window.get_window_type();
         actor._notifyWindowTypeSignalId =
@@ -1491,7 +1503,7 @@ var WindowManager = class {
             actor.show();
             this._mapping.add(actor);
 
-            actor.ease({
+            this._easeAfterOverview(actor, {
                 opacity: 255,
                 scale_x: 1,
                 scale_y: 1,
@@ -1508,7 +1520,7 @@ var WindowManager = class {
             actor.show();
             this._mapping.add(actor);
 
-            actor.ease({
+            this._easeAfterOverview(actor, {
                 opacity: 255,
                 scale_x: 1,
                 scale_y: 1,
diff --git a/js/ui/workspace.js b/js/ui/workspace.js
index 1a5bd02271ace33036ab1f7e2cd44cb9fdacb066..9ede6d6156da19bf4de387c3cf6ab5ecfb683aed 100644
--- a/js/ui/workspace.js
+++ b/js/ui/workspace.js
@@ -1225,7 +1225,8 @@ class Workspace extends St.Widget {
     }
 
     _windowAdded(metaWorkspace, metaWin) {
-        this._doAddWindow(metaWin);
+        if (!Main.overview.closing)
+            this._doAddWindow(metaWin);
     }
 
     _windowRemoved(metaWorkspace, metaWin) {
@@ -1233,7 +1234,7 @@ class Workspace extends St.Widget {
     }
 
     _windowEnteredMonitor(metaDisplay, monitorIndex, metaWin) {
-        if (monitorIndex == this.monitorIndex)
+        if (monitorIndex === this.monitorIndex && !Main.overview.closing)
             this._doAddWindow(metaWin);
     }
 
